src/sagemaker/pipeline.py

```diff
@@ -235,13 +254,94 @@ def _create_sagemaker_pipeline_model(self, instance_type):
         containers = self.pipeline_container_def(instance_type)
```

Comment:
I guess repack for the underlying models is captured here.

Reply:
Yes, the `pipeline_container_def` iterates all models and call each models' `prepare_container_def` under the hood. The `prepare_container_def` can capture the needs for runtime repack